src/Symfony/Component/Cache/Adapter/CouchbaseCollectionAdapter.php

```diff
@@ -183,7 +183,7 @@ protected function doSave(array $values, $lifetime): array|bool
         }
 
         $upsertOptions = new UpsertOptions();
-        $upsertOptions->expiry($lifetime);
+        $upsertOptions->expiry((new \DateTime())->modify('+'.$lifetime.' seconds'));
```

Comment:
Maybe we could just use this:

```php
$upsertOptions->expiry(new \DateTimeImmutable("+ $lifetime seconds"));
```

Reply:
Good idea, I updated this bit :slightly_smiling_face:

Reply:
Unfortunately this doesn't work. However, I updated to use a DateTimeImmutable  :+1: